node/silkworm/etl/buffer.hpp

```diff
@@ -46,18 +46,18 @@ class Buffer {
 
     void put(Entry&& entry) {
         // Add a new entry to the buffer
-        size_ += entry.size();
+        size_ += entry.size() + sizeof(head_t);
```

Comment:
shouldn't the same change be added to the other `put` overload? `void put(const Entry& entry)`